CMakeLists.txt

```diff
@@ -85,6 +85,16 @@ check_c_source_compiles("
   "
   HAVE_EXT_VECTOR_TYPE
 )
+check_c_source_compiles("
+    int main()
+    {
+        int a = 1;
+        __typeof__ (a) b = ({a + 1;});
```

Comment:
Now we check two features in one test, `__typeof__` and "statements and declarations in expressions". 
I like the old 27d2808 commit more, I doubt there will be other compilers that support all these extensions.